src/main/java/nuffle/storage/Storage.java

```diff
@@ -0,0 +1,127 @@
+package nuffle.storage;
+
+import nuffle.task.Deadline;
+import nuffle.task.Event;
+import nuffle.task.Task;
+import nuffle.task.Todo;
+
+import java.io.*;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Objects;
+
+/**
+ * Handles loading and saving tasks to and from a file.
+ * The file format is used to read and write tasks in a specific format.
+ */
+public class Storage {
+    // variable to store the file path
+    private final String filePath;
+
+    /**
+     * Constructs a Storage object with the specified file path.
+     *
+     * @param filePath The path to the file where tasks are stored.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads tasks from the file specified by the file path.
+     * The method reads the file line by line and creates appropriate Task objects based on the file's content.
+     *
+     * @return An ArrayList of tasks loaded from the file.
+     * @throws IOException If an I/O error occurs while reading the file.
+     */
+    public ArrayList<Task> load() throws IOException {
+        ArrayList<Task> tasks = new ArrayList<>();
+        File taskFile = new File(filePath);
+
+        // check if the file exists. If it does not, make the directory
+        if (!taskFile.exists()) {
+            if (taskFile.getParentFile().mkdir()) {
+                taskFile.createNewFile();
+                return tasks;
+            } else {
+                // throw an error if there is issue creating directory
+                throw new IOException("Problem creating the directory: " + taskFile.getParent());
+            }
+        }
+
+        // if the file exists, need to buffer the content
+        FileReader fr = new FileReader(filePath);
+        BufferedReader buffer = new BufferedReader(fr);
+
+        String eachLine = buffer.readLine();
+        while (eachLine != null) {
+            // dissect eachLine based on the expected format
+            // for example: D | 0 | return book | June 6th, then split it by "|"
+            String[] components = eachLine.split("\\|");
+
+            String category = components[0].trim();
+            String desc = components[2].trim();
```

Comment:
perhaps you might want to consider checking the size of the array before doing any operations on it in case of an IndexOutOfBoundsException should the save file be corrupted, e.g. missing a field like D | return book | June 6th